addons/block_code/code_generation/block_extension.gd

```diff
@@ -8,13 +8,28 @@
 ## property to draw information about the scene.[br]
 ## [br]
 ## To customize the user interface for a block, override public functions such
-## as [method get_defaults].
+## as [method get_defaults].[br]
+## [br]
+## In some cases, an extension may need to monitor the scene to determine if it
+## has changed. To achieve this, override [method _context_node_changed] to
+## connect the relevant signals, and call [method _emit_changed] when a
+## significant change has occurred.
 class_name BlockExtension
 extends RefCounted
 
+signal changed
```

Comment:
Oh wait I thought by looking at the `definition.changed.connect` in previous commit that you were going to use the `changed` signal already available in Resource https://docs.godotengine.org/en/stable/classes/class_resource.html#class-resource-method-emit-changed . But this is for the extension, got it.

Reply:
To be honest I thought that too, then I ended up doing that whole thing moving block extensions closer to Block, but then I left it because we probably _should_ connect to that changed signal as well. I really wish changed was in the Object class so it didn't feel weird copying Resource here. Some day maybe.